packages/query-core/src/queryObserver.ts

```diff
@@ -636,16 +636,23 @@ export class QueryObserver<
         return true
       }
 
+      if (!this.options.notifyOnChangeProps && !this.trackedProps.size) {
+        return true
+      }
+
       const { notifyOnChangeProps } = this.options
+      const notifyOnChangePropsValue =
+        typeof notifyOnChangeProps === 'function'
+          ? notifyOnChangeProps()
+          : notifyOnChangeProps
```

Comment:
hmm, why isn't it enough to move `const notifyOnChangePropsValue = ...` to the beginning, and then use `notifyOnChangePropsValue` everywhere instead of `notifyOnChangeProps`. That way, we could keep the one if statement like it was before:

```
      if (
        notifyOnChangePropsValue === 'all' ||
        (!notifyOnChangePropsValue && !this.trackedProps.size)
      ) {
        return true
      }
```

or am I missing something?

Reply:
No specific reason - changed to one single condition 👍